### PR TITLE
Update Makefile deploy_release to use jenkins.

### PR DESCRIPTION
**Pull request by jasonthomas**
I've updated deploy_release to use jenkins.

$JENKINS_USERNAME and $JENKINS_API_TOKEN env variables need to be set in order for it work correctly. JENKINS_API_TOKEN can be found at https://deploy.mktadm.ops.services.phx1.mozilla.com/me/configure -> 'Show API Tokens'.

r? @robhudson @oremj

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -69,21 +70,18 @@ tag_release: tagz.py
 	$(eval RELEASE_DATE := $(shell $(PYTHON) -c 'import datetime; now = datetime.datetime.utcnow(); tue = now + datetime.timedelta(days=(1 - now.weekday()) % 7); print tue.strftime("%Y.%m.%d")'))
 	$(PYTHON) tagz.py -r mozilla/solitude,mozilla/spartacus,mozilla/webpay,mozilla/commbadge,mozilla/fireplace,mozilla/marketplace-operator-dashboard,mozilla/marketplace-stats,mozilla/monolith-aggregator,mozilla/transonic,mozilla/zamboni -c create -t $(RELEASE_DATE)
 
-deploy_release:
+deploy_release: check_deploy_env
 	$(eval RELEASE_DATE := $(shell $(PYTHON) -c 'import datetime; now = datetime.datetime.utcnow(); tue = now + datetime.timedelta(days=(1 - now.weekday()) % 7); print tue.strftime("%Y.%m.%d")'))
-	$(PYTHON) scripts/dreadnot-deploy.py			\
```
Comment (robhudson) [inline Makefile]: since this isn't used perhaps delete that script? Or is it still useful?

File: Makefile [hunk 2]
```diff
@@ -69,21 +70,18 @@ tag_release: tagz.py
...
-		marketplace.allizom.org-webpay			\
-		monolith.allizom.org-aggregator			\
-		marketplace.allizom.org-marketplace-operator-dashboard	\
-		marketplace.allizom.org-marketplace-stats	\
-		marketplace.allizom.org-commbadge		\
-                marketplace.allizom.org-transonic               \
-		marketplace.allizom.org-fireplace		\
-		marketplace.allizom.org-zamboni
-
+	curl -k -X POST $(JENKINS_URL) \
```
Comment (robhudson) [inline Makefile]: Is the `-k` b/c of a home built cert? It'll be cool when Mozilla does that free SSL cert stuff they announced.

File: Makefile (hunk 1 above)
Comment (jasonthomas) [inline Makefile]: I don't think we need it anymore. I'll remove it.

File: Makefile (hunk 2 above)
Comment (jasonthomas) [inline Makefile]: Yes. We are using a Mozilla CA signed certificate for internal use.

**Comment by robhudson** [comment]
Looks awesome. r+

**Comment by oremj** [comment]
r+

**Pull request merged**